Mathlib/LinearAlgebra/FreeProduct/Basic.lean

```diff
@@ -116,37 +116,97 @@ abbrev FreeTensorAlgebra := TensorAlgebra R (⨁ i, A i)
 before taking the quotient by the free product relation.-/
 abbrev PowerAlgebra := ⨁ (n : ℕ), TensorPower R n (⨁ i, A i)
 
+/--The canonical linear injection from the direct sum of the `A i` to the power algebra.-/
+@[simp] noncomputable def PowerAlgebra.ι : (⨁ i, A i) →ₗ[R] PowerAlgebra R A :=
+  DirectSum.lof R ℕ (⨂[R]^· (⨁ i, A i)) 1 ∘ₗ
+    (MultilinearMap.ofSubsingleton R (⨁ i, A i) _ (0 : Fin 1)
+      |>.symm (PiTensorProduct.tprod R))
+
 /--The free tensor algebra and its representation as an infinite direct sum
 of tensor powers are (noncomputably) equivalent as `R`-algebras.-/
 @[reducible] noncomputable def powerAlgebra_equiv_freeAlgebra :
     PowerAlgebra R A ≃ₐ[R] FreeTensorAlgebra R A :=
   TensorAlgebra.equivDirectSum.symm
 
+theorem PowerAlgebra.ofDirectSum_ι :
+    (ofDirectSum.toLinearMap ∘ₗ PowerAlgebra.ι R A) = TensorAlgebra.ι R := by
+  ext i a
+  simp_rw [ι, LinearMap.comp_apply, lof_eq_of, MultilinearMap.ofSubsingleton_symm_apply_apply,
+            ofDirectSum.toLinearMap_apply, ofDirectSum_of_tprod, tprod_apply,
+           List.ofFn_const, List.replicate_one, List.prod_singleton]
+
 /--The generating equivalence relation for elements of the free tensor algebra
 that are identified in the free product.-/
 inductive rel : FreeTensorAlgebra R A → FreeTensorAlgebra R A → Prop
+  /--`rel` relates the injections of any `(1 : A i)` to `1`.-/
   | id  : ∀ {i : I}, rel (ι R <| lof R I A i 1) 1
+  /--`rel` relates the products of injections from the same `A i` to the injection of their
+  product in `A i`.-/
   | prod : ∀ {i : I} {a₁ a₂ : A i},
       rel
         (tprod R (⨁ i, A i) 2 (fun | 0 => lof R I A i a₁ | 1 => lof R I A i a₂))
         (ι R <| lof R I A i (a₁ * a₂))
 
-/--The generating equivalence relation for elements of the power algebra
-that are identified in the free product. -/
-@[reducible, simp] def rel' := rel R A on ofDirectSum
-
+-- We reproduce `rel.id` and `.prod here to make `R, A, i...` explicit.
```

Comment:
`rel.id` and `rel.prod` are automatically defined by the compiler when I define `rel` above; however, they aren't terribly ergonomic to use, as Lean automatically makes `R` and `A` implicit, and I can't figure out how to override that. Rather than fight the compiler too much, I just defined a second copy with more useful binders.